File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = 0, cnt = 0;
        for (int x : nums) {
            if (cnt == 0) {
                ans = x;
                cnt = 1;
            } else if (ans == x) {
                cnt++;
            } else {
                cnt--;
            }
        }
        return ans;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans,cnt = 0,0
        for x in nums:
            if cnt == 0:
                ans,cnt = x,1
            elif ans == x:
                cnt+=1
            else:
                cnt-=1
        return ans